tests/keras/engine/test_training.py

```diff
@@ -1044,6 +1049,14 @@ def test_target_tensors():
                                   'dense_b': target_b})
     model.train_on_batch(input_val, None)
 
+    # multi-output, not enough target tensors when `target_tensors` is not a dict
+    with pytest.raises(ValueError):
```

Comment:
When testing errors, please add a regex to check that the expected error messages is being printed (we don't do this everywhere now, but really we should). This is to avoid a situation where the test passes because *another* error of the same type is raised somewhere else (we ran into this issue several times before).